### ISSUE TITLE
9.0.x: nRF: AnalogIn.value always returns 0

### ISSUE THREAD

**Issue by dhalbert** [adafruit/circuitpython#9109]
Starting with early 9.0.0 releases, `AnalogIn` on nordic boards returns zero all the time, no matter what input is applied to the pin. For instance, even with `A0` tied to 3.3V, you see:

```py
>>> import analogio,board
>>> a0 = analogio.AnalogIn(board.A0)
>>> a0.value
0

```
The underlying code has not changed in some time.

We are currently using gcc13. I recompiled a `nordic` build 9.0.x with gcc10, and the problem goes away (!).

Thanks @caternuson for finding this.

**Comment by dhalbert** [comment]
Fixed by #9114.

**Comment by kevinjwalters** [comment]
I was just scratching my head over this in 9.0.2 and was about to try 8.x then found this!

Digital inputs are kaput too https://github.com/adafruit/circuitpython/issues/9154
